rust_src/src/lists.rs

```diff
@@ -142,8 +152,8 @@ impl Iterator for TailsIter {
     fn next(&mut self) -> Option<Self::Item> {
         match self.tail.as_cons() {
             None => {
```

Comment:
This is where the existing code is wrong. No error should be raised unless requested.